### PR TITLE
Add test for duplicate path url but different methods

### PR DESCRIPTION
**Pull request by robd003**
Test for issue https://github.com/sanic-org/sanic-routing/issues/43

### CODE REVIEW

File: tests/test_routing.py [hunk 1]
```diff
@@ -429,3 +429,24 @@ def handler2():
...
+
+    def handler2():
+        return "handler2"
+
+    router = Router()
+    router.add("/<foo:path>", handler1, methods=["GET", "OPTIONS"])
+    router.add("/<foo:path>", handler2, methods=["POST"])
+    router.finalize()
+
+    _, handler, params = router.get("/a-random-path", "GET")
```
Comment (ahopkins) [inline tests/test_routing.py:446]: Since we're doing path, can you make the test `/a/random/path`? Or perhaps even better, maybe parametrize the test and do both a single and multi segment path.

**Review by ahopkins — CHANGES_REQUESTED** [review]
thanks for this. Test looks good, just one small change.

**Comment by robd003** [comment]
@ahopkins Let me know if this is good

File: tests/test_routing.py (hunk 1 above)
Comment (robd003) [reply tests/test_routing.py]: @ahopkins Just pushed an update. Let me know if it works for you.

**Comment by robd003** [comment]
For the routes is it worth marking which methods they support and then when searching for paths to limit the search for current method only?

In the test case above I was seeing a <path> matcher overtake an <int> even though the <path> wasn't supposed to respond for the method being requested.

**Comment by ahopkins** [comment]
I am not sure I understand the question. The issue is (as you said) that the last route definition is taking priority for the methods

**Comment by ahopkins** [comment]
Replaced by #45
